### PR TITLE
Set multiple GenericContam controllers correctly 

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
Multiple GenericContam controllers are not set correctly. This fix correct controller setup.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: tst/EnergyPlus/unit/ZoneContaminantPredictorCorrector.unit.cc [hunk 1]
```diff
@@ -390,3 +390,210 @@ TEST_F(EnergyPlusFixture, ZoneContaminantPredictorCorrector_CorrectZoneContamina
...
+    DataContaminantBalance::ContaminantControlledZone(3).AvaiSchedPtr = 1;
+    DataContaminantBalance::ContaminantControlledZone(3).ActualZoneNum = 3;
+    DataContaminantBalance::ContaminantControlledZone(3).NumOfZones = 1;
+
+    PredictZoneContaminants(ShortenTimeStepSys, UseZoneTimeStepHistory, PriorTimeStep);
+    EXPECT_NEAR(19.549478386, DataContaminantBalance::GCPredictedRate(1), 0.00001);
+    EXPECT_NEAR(20.887992514, DataContaminantBalance::GCPredictedRate(2), 0.00001);
+    EXPECT_NEAR(21.251538064, DataContaminantBalance::GCPredictedRate(3), 0.00001);
+
+    DataContaminantBalance::Contaminant.CO2Simulation = false;
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ZoneContaminantPredictorCorrector.unit.cc]: Same comment here about moving this "reset" type statements into the clear_state function in DataContaminantBalance.cc.

File: src/EnergyPlus/ZoneContaminantPredictorCorrector.cc [hunk 2]
```diff
@@ -1755,23 +1755,37 @@ namespace ZoneContaminantPredictorCorrector {
...
-                        if (ContaminantControlledZone(ContControlledZoneNum).EMSOverrideGCSetPointOn) {
-                            ZoneAirGCSetPoint = ContaminantControlledZone(ContControlledZoneNum).EMSOverrideGCSetPointValue;
+                    if (ContaminantControlledZone(ContControlledZoneNum).ActualZoneNum == ZoneNum) {
+                        if (GetCurrentScheduleValue(ContaminantControlledZone(ContControlledZoneNum).AvaiSchedPtr) > 0.0) {
+                            ZoneAirGCSetPoint = ZoneGCSetPoint(ContaminantControlledZone(ContControlledZoneNum).ActualZoneNum);
+                            if (ContaminantControlledZone(ContControlledZoneNum).EMSOverrideCO2SetPointOn) {
+                                ZoneAirGCSetPoint = ContaminantControlledZone(ContControlledZoneNum).EMSOverrideGCSetPointValue;
+                            }
+                            ControlledGCZoneFlag = true;
+                            break;
```
Comment (Myoldmopar) [inline src/EnergyPlus/ZoneContaminantPredictorCorrector.cc:1781]: Same comment here about the break statement.  You do, definitely, want to break here, and not continue looping, right?

**Review by Myoldmopar — CHANGES_REQUESTED** [review]
Very minor notes that are exactly the same as my notes in #6745.  Once these minor things are addressed/discussed, these should drop in together.

File: src/EnergyPlus/ZoneContaminantPredictorCorrector.cc (hunk 2 above)
Comment (lgu1234) [reply src/EnergyPlus/ZoneContaminantPredictorCorrector.cc:1781]: @Myoldmopar Yes.

File: tst/EnergyPlus/unit/ZoneContaminantPredictorCorrector.unit.cc (hunk 1 above)
Comment (lgu1234) [reply tst/EnergyPlus/unit/ZoneContaminantPredictorCorrector.unit.cc]: @Myoldmopar I am going to move these reset statement into the clear_state function.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
When I merged in the CO2 branch, it caused a conflict in the unit test.  It was just Git trying to be very smart and seeing so much similarities in the two separate new unit tests.  I think I got them all resolved and building, so if it's all green on the next round, this will go in next.

**Pull request merged**